### DISCUSSION TITLE
Running C# compiled to WASM dlls in WASM runtimes

### DISCUSSION THREAD

**Discussion by schaveyt** [elringus/bootsharp#27] [Q&A]
With projects like [wasmedge runtime](https://github.com/WasmEdge/WasmEdge) does this library help enable compiling c# to WASM dlls executable by such runtimes?

**Comment by elringus** [comment]
The actual compilation to WASM is handled by .NET (https://github.com/dotnet/runtime/tree/main/src/mono/wasm).

This project main goals are:
1. Make it possible to use the compiled libraries in any JavaScript environment (.NET currently supports web browsers only)
2. Provide a user-friendly workflow for consuming the compiled libraries in JS: packing everything into single-file UMD library on publish, generating boilerplate interop code in both C# and JS, generating TypeScript definitions for the bindings, etc.

**Reply by schaveyt** [reply]
For point 1, is that the same as saying.."runs in Nodejs"?

**Reply by elringus** [reply]
There are other cases. For example, web extensions for VS Code, which can't use any node modules or window/document browser APIs. That's my primary use case for this project at the moment.
